src/js/client/renderers/BoardRenderer.js

```diff
@@ -161,31 +161,44 @@ export default class BoardRenderer {
   }
 
   renderTerrain(game) {
-    const terrainColor = {
-      normal: 0xFFFFFF,
+    const terrain = game.pathHelper.weights.terrain,
+          terrainColor = {
       crater: 0xCECECE,
       ridge: 0xE5E5E5,
       obstacle: 0x000000,
     }
-    let terrainContainer = new PIXI.Container()
+
+    let terrainContainer, terrainBackground
+    if (terrain.rendered === undefined) {
+      terrainContainer = new PIXI.Container()
+      terrainBackground = new PIXI.Graphics()
+      terrain.rendered = {
+        container: terrainContainer,
+        background: terrainBackground,
+      }
+      terrainContainer.addChild(terrainBackground)
+      terrainContainer.parentLayer = this.mapLayer
+      this.app.stage.addChild(terrainContainer)
+    } else {
+      terrainContainer = terrain.rendered.container
+      terrainBackground = terrain.rendered.background
+    }
+
+    terrainBackground.clear()
     for (let x = 0; x < game.width / GRID_SIZE; x++) {
       for (let y = 0; y < game.height / GRID_SIZE; y++) {
-        let terrainBackground = new PIXI.Graphics()
-        terrainBackground.position = {
+        let type = terrain.typeAt(x, y)
+        if (type === "normal") continue
```

Comment:
I'm trying to avoid relying on the JS feature of having an if statement without curly braces. Just throw curly braces around the `continue`.

    if (type === "normal") { continue }